src/main/scala/uk/gov/nationalarchives/downloadfiles/FileUtils.scala

```diff
@@ -43,6 +46,23 @@ class FileUtils()(implicit val executionContext: ExecutionContext, keycloakDeplo
       filePath <- pathFromResponse(response, fileId)
     } yield filePath
 
+  def addFileStatus(keycloakUtils: KeycloakUtils, client: GraphQLClient[afs.Data, afs.Variables], fileId: UUID, lambdaConfig: Map[String, String])(implicit backend: SttpBackend[Identity, Any]): Future[AddFileStatus] = {
+
+    val statusType = "Upload"
+    val statusValue = "Success"
+    for {
+      token <- keycloakUtils.serviceAccountToken(lambdaConfig("auth.client.id"), lambdaConfig("auth.client.secret"))
+      response <- client.getResult(token, document, Option(afs.Variables(AddFileStatusInput(fileId, statusType, statusValue))))
+      fileStatus <- getFileStatusResponse(response, statusType, statusValue)
+    } yield fileStatus
+  }
+
+  private def getFileStatusResponse(response: GraphQlResponse[afs.Data], statusType: String, statusValue: String): Future[AddFileStatus] = response.errors match {
+    case Nil => Future(response.data.get.addFileStatus)
+    case List(authorisedError: NotAuthorisedError) => failed(authorisedError.message)
+    case errors => failed(s"Unable to add file status with statusType '$statusType' and value '$statusValue'. Errors: ${errors.map(e => e.message).mkString}")
```

Comment:
Might be worth adding the file id into the error message to make tracking down any issues easier.

Reply:
Yeah, make sence, thanks